etcdserver/v3_server.go

```diff
@@ -143,11 +147,17 @@ func (s *EtcdServer) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse
 			warnOfExpensiveReadOnlyTxnRequest(s.getLogger(), start, r, resp, err)
 		}(time.Now())
 
-		get := func() { resp, err = s.applyV3Base.Txn(r) }
-		if serr := s.doSerialize(ctx, chk, get); serr != nil {
-			return nil, serr
+		get := func() error {
+			resp, err = s.applyV3Base.Txn(r)
+			if err != nil {
```

Comment:
While I agree that we should handle error, why don't you just `return err`? Later we are checking whether it's nil.

Reply:
your advice is ok.

Reply:
Thanks for making changes above but you didn't make any change here?